### PR TITLE
Improve the docs on creating Secrets

### PR DESCRIPTION
**Pull request by IwonaLanger**
**Description**

Changes proposed in this pull request:

- Add "Create a Service Instance with a Namespace-Based Secret"
- Enhance "Create a Service Instance with a Custom Secret"
- Add the diagram of Secrets Precedence
- Move task docs to the `user` folder
- Minor fixes

**Related issue(s)**
[#5875](https://github.tools.sap/kyma/backlog/issues/5875)

### CODE REVIEW

File: docs/assets/secrets_precedence_4.drawio.svg
Comment (grego952) [inline docs/assets/secrets_precedence_4.drawio.svg:1]: Change BTB to BTP

File: docs/user/03-20-multitenancy.md [hunk 1]
```diff
@@ -2,15 +2,28 @@
 
 With the SAP BTP Operator module, you can create configurations for several subaccounts in a single Kyma cluster.
 
+## Context
+
+By default, a Kyma cluster is associatted with one subaccount. Consequently, any service instance created within any namespace is provisioned in the associated subaccount. See [Preconfigured Credentials and Access](03-10-preconfigured-secret.md). However, SAP BTP Operator also supports configurations for several subaccounts in a single Kyma cluster.
```
Comment (grego952) [inline docs/user/03-20-multitenancy.md:7]:
```suggestion
By default, a Kyma cluster is associated with one subaccount. Consequently, any service instance created within any namespace is provisioned in the associated subaccount. See [Preconfigured Credentials and Access](03-10-preconfigured-secret.md). However, SAP BTP Operator also supports configurations for several subaccounts in a single Kyma cluster.
```

File: docs/user/03-21-create-service-instance-with-custom-secret.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+To create a service instance with a custom Secret, you must use the **btpAccessCredentialsSecret** field in the `spec` of the service instance. In it, you pass the Secret from the `kyma-system` namespace to create your service instance. You can use different Secrets for different service instances.
+
+## Prerequisites
+
+* A subaccount in the SAP BTP cockpit
+
+## Procedure
+
+### Create Your Custom Secret
```
Comment (grego952) [inline docs/user/03-21-create-service-instance-with-custom-secret.md:15]: In general, I'm missing where and how users should create service instances, bindings. Should it be done from the cockpit, dashboard? Somewhere else?

File: docs/user/03-22-create-service-instance-with-namespace-based-secret.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+To have service instances from one subaccount associated with one namespace, you must use a Secret dedicated to this namespace to create these service instances.
+
+## Prerequisites
+
+* A subaccount in the SAP BTP cockpit
+
+## Procedure
+
+### Create a Namespace-Based Secret
```
Comment (grego952) [inline docs/user/03-22-create-service-instance-with-namespace-based-secret.md:11]: See my previous comment

File: docs/user/03-21-create-service-instance-with-custom-secret.md (hunk 2 above)
Comment (IwonaLanger) [reply docs/user/03-21-create-service-instance-with-custom-secret.md:15]: In lines 83-84, there are links to instructions on creating SIs and SBs. Do you think I should move them up to the procedure?

File: docs/user/03-22-create-service-instance-with-namespace-based-secret.md (hunk 3 above)
Comment (IwonaLanger) [reply docs/user/03-22-create-service-instance-with-namespace-based-secret.md:11]: Same as above (lines 73-74).

File: docs/user/03-21-create-service-instance-with-custom-secret.md (hunk 2 above)
Comment (grego952) [reply docs/user/03-21-create-service-instance-with-custom-secret.md:15]: Or you can add inline links. For example: Create an [SAP Service Manager service instance](https://github.com/kyma-project/btp-manager/blob/f4021bb32ab14704ae4dee398cd9eb1bb536cf42/docs/user/03-30-management-of-service-instances-and-bindings.md#create-a-service-instance) with the `service-operator-access` plan

File: docs/user/03-22-create-service-instance-with-namespace-based-secret.md (hunk 3 above)
Comment (grego952) [reply docs/user/03-22-create-service-instance-with-namespace-based-secret.md:11]: See my previous comment :)

**Review by grego952 — APPROVED** [review]

**Review by jaroslaw-pieszka — APPROVED** [review]

**Pull request merged**